tests/e2e/vmlistener_test.go

```diff
@@ -120,7 +124,7 @@ func (s *VMListenerTestParams) TestKillVM(c *C) {
 
 	// power on vm
 	govc.PowerOnVM(s.dockerHostName)
-	misc.SleepForSec(35)
+	misc.SleepForSec(45)
```

Comment:
I have tried the poller and with following sequence needed to rely on the static time out

1. vm is getting powered on within 10 seconds (able to ping the vm but VM is not in fully powered mode)
2. docker daemon takes some time to respond and it is variable
3. even thought docker is available there is no way to find out docker has finished its background processing
4. vDVS plugin refcount thread is running in background and no hook to hang on reliably 

Ideally would like to have poller for step 1 & 4. Step 1 is possible but not 2/3/4.

Let me know what do you think.